src/components/Api.js

```diff
@@ -0,0 +1,118 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    (this._baseUrl = baseUrl), (this._headers = headers);
+  }
+
+  async validateAndParse(res) {
+    if (res.ok) {
+      return await res.json();
+    } else {
+      return Promise.reject();
+    }
+  }
+
+  async getInitialCards() {
+    try {
+      const res = await fetch(`${this._baseUrl}/cards`, {
+        headers: this._headers,
+      });
+      return this.validateAndParse(res);
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  async getUserInfo() {
+    try {
+      const res = await fetch(`${this._baseUrl}/users/me`, {
+        headers: this._headers,
+      });
+      return this.validateAndParse(res);

```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this.validateAndParse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.